openmc/config.py

```diff
@@ -0,0 +1,83 @@
+from collections.abc import MutableMapping
+import os
+from pathlib import Path
+import warnings
+
+from openmc.data import DataLibrary
+
+__all__ = ["config"]
+
+
+class _Config(MutableMapping):
+    def __init__(self, data=()):
+        self._mapping = {}
+        self.update(data)
+
+    def __getitem__(self, key):
+        return self._mapping[key]
+
+    def __delitem__(self, key):
+        del self._mapping[key]
+        if key == 'cross_sections':
+            del os.environ['OPENMC_CROSS_SECTIONS']
+        elif key == 'mg_cross_sections':
+            del os.environ['OPENMC_MG_CROSS_SECTIONS']
+
+    def __setitem__(self, key, value):
+        if key == 'cross_sections':
+            # Force environment variable to match
+            self._set_path(key, value)
+            os.environ['OPENMC_CROSS_SECTIONS'] = str(value)
```

Comment:
Is it worth checking if the value ends with ".xml" here

Reply:
It's not strictly required -- hypothetically one could name their cross sections XML file something like `endf71.xs` if they really wanted